### PR TITLE
use precomputed tables for ed25519 and secp256k1

### PR DESCRIPTION
**Pull request by polubelova**
This PR is a follow-up to #604. 

**Future work**: fix the [issue](https://github.com/hacl-star/hacl-star/pull/604#pullrequestreview-1120949625) with `seq_of_list` so that we can generate a table of a larger size. Such a table will be needed when we have support for EC multi-scalar multiplication with different window sizes (to improve the performance of `ecdsa-verify`).

### CODE REVIEW

File: code/bignum/Hacl.Spec.PrecompBaseTable.fst [hunk 1]
```diff
@@ -0,0 +1,226 @@
...
+module LSeq = Lib.Sequence
+
+module LE = Lib.Exponentiation
+module SE = Spec.Exponentiation
+module BE = Hacl.Impl.Exponentiation.Definitions
+
+#set-options "--z3rlimit 50 --fuel 0 --ifuel 0"
+
+#push-options "--fuel 2"
+val list_append_index1: #a:Type -> x:list a -> y:list a -> i:nat{i < FL.length x} ->
```
Comment (R1kM) [inline code/bignum/Hacl.Spec.PrecompBaseTable.fst:16]: Would it be worth adding these lemmas to F*'s ulib/ ?

**Review by R1kM — APPROVED** [review]
Looks good, thanks!

**Comment by karthikbhargavan** [comment]
Are there some performance benchmarks for this?

**Comment by msprotz** [comment]
@karthikbhargavan approve this?

**Review by karthikbhargavan — APPROVED** [review]

**Comment by polubelova** [comment]
I'll update this PR by setting window size `w` to 5 for `ecdsa-verify`. It will improve the performance by ~6% (compared to `main`).

Here is the number of `point_add` and `point_double` needed for `ecdsa-verify`:
```
 w = 3
 # point add = 174
 # point double = 258
 
 if point double = 0.8 * point add, # point add ~= 380
 if point double = 0.6 * point add, # point add ~= 328
 
 w = 4
 # point add = 135
 # point double = 263
 
 if point double = 0.8 * point add, # point add ~= 345
 if point double = 0.6 * point add, # point add ~= 292
 
 w = 5
 # point add = 118
 # point double = 270
 
 if point double = 0.8 * point add, # point add ~= 334
 if point double = 0.6 * point add, # point add ~= 280
 
 w = 6
 # point add = 116
 # point double = 283
 
 if point double = 0.8 * point add, # point add ~= 342
 if point double = 0.6 * point add, # point add ~= 285
 
 w = 7
 # point add = 136
 # point double = 315
 
 if point double = 0.8 * point add, # point add ~= 388
 if point double = 0.6 * point add, # point add ~= 325
```
For the current implementation, `w = 5` is the best option. Just need to move two precomputed tables for a base point to a separate file (`w = 4` for `ecdsa-sign` and `w = 5` for `ecdsa-verify`).

**Pull request merged**